mllib/src/main/scala/org/apache/spark/mllib/clustering/LDA.scala

```diff
@@ -311,165 +319,319 @@ private[clustering] object LDA {
 
   private[clustering] type TokenCount = Double
 
-  /** Term vertex IDs are {-1, -2, ..., -vocabSize} */
-  private[clustering] def term2index(term: Int): Long = -(1 + term.toLong)
 
-  private[clustering] def index2term(termIndex: Long): Int = -(1 + termIndex).toInt
 
-  private[clustering] def isDocumentVertex(v: (VertexId, _)): Boolean = v._1 >= 0
+  object LearningAlgorithms extends Enumeration {
+    type Algorithm = Value
+    val Gibbs, EM = Value
+  }
 
-  private[clustering] def isTermVertex(v: (VertexId, _)): Boolean = v._1 < 0
+  private[clustering] trait LearningState {
+    def next(): LearningState
+    def topicsMatrix: Matrix
+    def describeTopics(maxTermsPerTopic: Int): Array[(Array[Int], Array[Double])]
```

Comment:
Why is it not necessary? The LDASuite which contains the Distributed/LocalModels calls it. How they are created, is up to the specific implementation of LDA. Could you be more specific why its not necessary?